src/test/java/com/google/cloud/tools/gradle/endpoints/framework/TestProject.java

```diff
@@ -63,20 +69,31 @@ public BuildResult build() throws URISyntaxException, IOException {
     if (hostname != null) {
       injectHostname(hostname);
     }
+    if (basePath != null) {
+      injectBasePath(basePath);
+    }
     return GradleRunner.create()
         .withProjectDir(testDir)
         .withPluginClasspath()
         .withArguments(gradleRunnerArgs)
         .build();
   }
 
-  // inject an endpoints plugin hostname into the pom.xml
+  // Inject an endpoints plugin hostname into the build.gradle file.
   private void injectHostname(String hostname) throws IOException {
+    injectIntoBuildGradleFile("/\\*endpoints-plugin-hostname\\*/", "endpointsServer.hostname = '" + hostname + "'");
+  }
+
+  // Inject an endpoints plugin basePath into the build.gradle file.
+  private void injectBasePath(String basePath) throws IOException {
+    injectIntoBuildGradleFile("/\\*endpoints-plugin-basePath\\*/", "endpointsServer.basePath = '" + basePath + "'");
```

Comment:
Should endpoints-plugin-basePath be all lowercase? (endpoints-plugin-base**p**ath)

Reply:
so that's actually a part of the test project, we're just doing a string replace, see: https://github.com/GoogleCloudPlatform/endpoints-framework-gradle-plugin/blob/master/src/test/resources/projects/server/build.gradle#L48 for the string we're replacing for hostname.

Reply:
you have to add another comment to allow the replacement of that string.